src/richie/plugins/lti_consumer/models.py

```diff
@@ -0,0 +1,119 @@
+"""
+LTI consumer plugin models
+"""
+from urllib.parse import unquote
+
+from django.conf import settings
+from django.db import models
+from django.utils.translation import gettext_lazy as _
+
+import exrex
+from cms.models.pluginmodel import CMSPlugin
+from cms.utils import get_current_site
+from oauthlib import oauth1
+
+from .defaults import INSTRUCTOR, STUDENT
+
+
+class LtiConsumer(CMSPlugin):
+    """
+    LTI consumer plugin model.
+    """
+
+    url = models.URLField(_("LTI url"), null=True, blank=True)
+    context_id = models.CharField(max_length=100, null=True, blank=True)
+    lti_provider_id = models.CharField(max_length=50, null=True, blank=True)
+    oauth_consumer_key = models.CharField(max_length=50, null=True, blank=True)
+    shared_secret = models.CharField(max_length=50, null=True, blank=True)
+
+    # pylint: disable=signature-differs
+    def save(self, *args, **kwargs):
+        """
+        Initialize attributes :
+        - context_id from current site domain
+        - url if predefined provider is used
+        """
+        if not self.id:
+            site = get_current_site()
+            self.context_id = site.domain
+            if self.lti_provider and not self.url:
+                self.url = "{:s}".format(
+                    exrex.getone(self.lti_provider.get("base_url")),
+                )
+        super().save(*args, **kwargs)
+
+    @staticmethod
+    def available_providers():
+        """
+        Returns available predefined LTI providers settings
+        """
+        return [
+            (provider_key, provider.get("display_name"))
```

Comment:
Avoid an empty display name?
```suggestion
    def available_providers(self):
        """
        Returns available predefined LTI providers settings
        """
        return [
            (provider_key, provider.get("display_name", self.lti_provider_id))
```

Reply:
Ok for avoiding empty name, but making this function non static will cause problems, as it is used before instantiation.

Reply:
This function has been removed now.